CMakeLists.txt

```diff
@@ -1,39 +1,51 @@
 cmake_minimum_required(VERSION 3.15 FATAL_ERROR)
+
+include(gen/CurrentBuildVulkanVersion.cmake)
+
 project(VulkanBootstrap
     LANGUAGES CXX
-    DESCRIPTION "A Vulkan utility library to ease the initialization steps in Vulkan")
+    DESCRIPTION "A Vulkan utility library to ease the initialization steps in Vulkan"
+    VERSION ${VK_BOOTSTRAP_SOURCE_HEADER_VERSION})
 
 option(VK_BOOTSTRAP_DISABLE_WARNINGS "Disable warnings during compilation" OFF)
 option(VK_BOOTSTRAP_WERROR "Enable warnings as errors during compilation" OFF)
 
-add_library(vk-bootstrap-vulkan-headers INTERFACE)
-
 set(VK_BOOTSTRAP_VULKAN_HEADER_DIR "" CACHE STRING "Specify the location of the Vulkan-Headers include directory.")
 mark_as_advanced(VK_BOOTSTRAP_VULKAN_HEADER_DIR)
 
-include(gen/CurrentBuildVulkanVersion.cmake)
-
+# Check if the user has set this variable explicitly
 if(NOT "${VK_BOOTSTRAP_VULKAN_HEADER_DIR}" STREQUAL "")
-    target_include_directories(vk-bootstrap-vulkan-headers INTERFACE $<BUILD_INTERFACE:${VK_BOOTSTRAP_VULKAN_HEADER_DIR}>)
-else ()
-    find_package(Vulkan QUIET)
-
-    if(${Vulkan_INCLUDE_DIR} STREQUAL "Vulkan_INCLUDE_DIR-NOTFOUND")
-        include(FetchContent)
-        FetchContent_Declare(
-            VulkanHeaders
-            GIT_REPOSITORY https://github.com/KhronosGroup/Vulkan-Headers
-            GIT_TAG        ${VK_BOOTSTRAP_SOURCE_HEADER_VERSION_GIT_TAG}
-        )
-        FetchContent_MakeAvailable(VulkanHeaders)
-        target_link_libraries(vk-bootstrap-vulkan-headers INTERFACE Vulkan::Headers)
-    else()
-        set_target_properties(vk-bootstrap-vulkan-headers PROPERTIES INTERFACE_INCLUDE_DIRECTORIES ${Vulkan_INCLUDE_DIR})
+    add_library(Vulkan-Headers INTERFACE)
+    target_include_directories(Vulkan-Headers INTERFACE $<BUILD_INTERFACE:${VK_BOOTSTRAP_VULKAN_HEADER_DIR}>)
+
+# Check if the target is already defined
+elseif(NOT (TARGET Vulkan-Headers OR TARGET Vulkan::Headers))
+    # Try looking for the Vulkan-Headers package directly
+    find_package(VulkanHeaders CONFIG)
+    if (NOT VulkanHeaders_FOUND)
+        # Try looking using the CMake built in Vulkan support
+        find_package(Vulkan QUIET)
+        if(NOT Vulkan_FOUND)
+            # Lastly just grab Vulkan-Headers directly using FetchContent
+            include(FetchContent)
+            FetchContent_Declare(
+                Vulkan-Headers-for-vk-bootstrap
+                GIT_REPOSITORY https://github.com/KhronosGroup/Vulkan-Headers
+                GIT_TAG        ${VK_BOOTSTRAP_SOURCE_HEADER_VERSION_GIT_TAG}
+            )
+            FetchContent_MakeAvailable(Vulkan-Headers-for-vk-bootstrap)
+        endif()
     endif()
 endif()
+# find_package(Vulkan) only defines Vulkan::Headers, create Vulkan-Headers in this case
+if(TARGET Vulkan::Headers AND NOT TARGET Vulkan-Headers)
```

Comment:
Why do you want the Vulkan-Headers target instead of Vulkan::Headers?

Reply:
Vulkan::Headers cannot be installed because its an alias, and using `Vulkan-Headers` everywhere simplifies the search logic since I only have to look for 1 target. Like, if Vulkan::Headers was found through `find_package(Vulkan)` the target isn't an alias, but if I use `find_package(VulkanHeaders)`, it *is* an alias (and the install code wasn't happy).

Reply:
> Vulkan::Headers cannot be installed because its an alias

You shouldn't be installing Vulkan::Headers

Reply:
Goes to show what all I know about cmake & installing things! I'll fix it up shortly.